spark/src/main/scala/org/apache/spark/sql/comet/CometScanExec.scala

```diff
@@ -490,8 +490,7 @@ object CometScanExec extends DataTypeSupport {
       // TODO add map
       dt match {
         case s: StructType => s.fields.map(_.dataType).forall(isTypeSupported)
-        // TODO: Add nested array and iceberg compat support
-        // case a: ArrayType => isTypeSupported(a.elementType)
+        case a: ArrayType => isTypeSupported(a.elementType)
```

Comment:
Are we sure about this?

Reply:
I'm not sure myself how well arrays are supported by the scans. I suppose we need to add more tests for arrays?

Reply:
Thanks @parthchandra I'm planning to merge it and add more array tests as followup PR